### PR TITLE
Test with Java 21

### PR DESCRIPTION
**Pull request by gounthar**
Java 21 was released Sep 19, 2023. We want to announce full support for Java 21 in early October and would like the most used plugins to be compiled and tested with Java 21.

The acceptance test harness and plugin bill of materials tests are already passing with Java 21. This is a further step to improve plugin readiness for use with Java 21 and for development with Java 21.

The change intentionally tests only two Java configurations, Java 17 and Java 21 because we believe that the risk of a regression that only affects Java 11 is shallow. We generate Java 11 byte code with the Java 17 and the Java 21 builds, so we're already testing Java 11 byte code.

Testing done
===

Confirmed tests pass with Java 21 on Linux.

### Submitter checklist
- [X] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [X] Ensure that the pull request title represents the desired changelog entry
- [X] Please describe what you did
- ~~Link to relevant issues in GitHub or Jira~~
- ~~Link to relevant pull requests, esp. upstream and downstream changes~~
- [X] Ensure you have provided tests - that demonstrates feature works or fixes the issue

### CODE REVIEW

**Comment by bguerin** [comment]
Thanks for your work, but switching to java 21 is already done and merged:
https://github.com/jenkinsci/pipeline-maven-plugin/pull/719

File: Jenkinsfile [hunk 1]
```diff
@@ -5,7 +5,7 @@ buildPlugin(
   // cannot use this with Docker tests
   useContainerAgent: false,
   configurations: [
-    [ platform: "linux", jdk: "21" ],
-    [ platform: "windows", jdk: "17" ]
+    [ platform: "linux", jdk: 21 ],
+    [ platform: "windows", jdk: 17 ]
```
Comment (bguerin) [inline Jenkinsfile:9]: I know that the README of the jenkinsci pipeline library says exactly that, but jdk versions are String:
https://github.com/jenkins-infra/pipeline-library/blob/master/vars/buildPlugin.groovy#L33

**Comment by gounthar** [comment]
> Thanks for your work, but switching to Java 21 is already done and merged: #719

Sorry about that. I started the work yesterday and hadn't looked at the open PR. 🤷 
I have a long list of plugins to migrate; I should add in my workflow prerequisites "See if the maintainer has already done the work". 🤣

File: Jenkinsfile (hunk 1 above)
Comment (gounthar) [reply Jenkinsfile:9]: As for the content of the file, I was (blindly?) following the [official archetype](https://github.com/jenkinsci/archetypes/blob/master/common-files/Jenkinsfile). 🤷

**Comment by gounthar** [comment]
> Thanks for your work, but switching to Java 21 is already done and merged: #719

Sorry about that. I started the work yesterday and hadn't looked at the open PR. 🤷 
I have a long list of plugins to migrate; I should add in my workflow prerequisites "See if the maintainer has already done the work". 🤣

**Comment by bguerin** [comment]
> > Thanks for your work, but switching to Java 21 is already done and merged: #719
> 
> Sorry about that. I started the work yesterday and hadn't looked at the open PR. 🤷 I have a long list of plugins to migrate; I should add in my workflow prerequisites "See if the maintainer has already done the work". 🤣

No worry. It's me who is sorry, for your spent time !

**Pull request closed without merging**